roughpy_jax/src/cpu/dense_ft_fma.cpp

```diff
@@ -0,0 +1,193 @@
+#include "cpu/dense_ft_fma.hpp"
+
+#include <cstring>
+#include <type_traits>
+
+#include "roughpy_compute/dense/basic/free_tensor_fma.hpp"
+
+namespace {
+
+namespace ffi = xla::ffi;
+
+// FIXME split into 32 and 64 variants with templates?
+using RpyFloatType = float;
+inline constexpr ffi::DataType XlaIndexType = ffi::DataType::S32;
+inline constexpr ffi::DataType XlaFloatType = ffi::DataType::F32;
+static_assert(
+    std::is_same_v<
+        RpyFloatType,
+        ffi::NativeType<XlaFloatType>
+    >,
+    "XlaFloatType must match underlying float type"
+);
```

Comment:
We need a nice mechanism for dealing with this and handling whether or not we include 64-bit builds for JAX.